python/python/tests/tests_server_modules/test_ft.py

```diff
@@ -198,3 +199,85 @@ async def _create_test_index_with_vector_field(
+        assert b"Field {\n  price\n  0\n  10\n}\n" == result
+
+        # FT.EXPLAIN on a search query containing numeric field and having bytes type input to the command.
+        result = await ft.explain(
+            glide_client, indexName=indexName.encode(), query=query.encode()
+        )
+        assert b"Field {\n  price\n  0\n  10\n}\n" == result
+
+        # FT.EXPLAIN on a search query that returns all data.
+        result = await ft.explain(glide_client, indexName=indexName, query="*")
+        assert b"*\n" == result
+
+        assert await ft.dropindex(glide_client, indexName=indexName)
+
+        # FT.EXPLAIN on a missing index throws an error.
+        with pytest.raises(RequestError):
+            await ft.explain(glide_client, str(uuid.uuid4()), "*")
+
+    @pytest.mark.parametrize("cluster_mode", [True])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_ft_explaincli(self, glide_client: GlideClusterClient):
+        indexName = str(uuid.uuid4())
+        await TestFt._create_test_index_for_ft_explain_commands(
+            self=self, glide_client=glide_client, index_name=indexName
+        )
+
+        # FT.EXPLAINCLI on a search query containing numeric field.
+        query = "@price:[0 10]"
+        result = await ft.explaincli(glide_client, indexName=indexName, query=query)
+        assert [b"Field {", b"  price", b"  0", b"  10", b"}", b""] == result
+
+        # FT.EXPLAINCLI on a search query containing numeric field and having bytes type input to the command.
+        result = await ft.explaincli(
+            glide_client, indexName=indexName.encode(), query=query.encode()
+        )
+        assert [b"Field {", b"  price", b"  0", b"  10", b"}", b""] == result
+
+        # FT.EXPLAINCLI on a search query that returns all data.
+        result = await ft.explaincli(glide_client, indexName=indexName, query="*")
+        assert [b"*", b""] == result
+
+        assert await ft.dropindex(glide_client, indexName=indexName)
+
+        # FT.EXPLAINCLI on a missing index throws an error.
+        with pytest.raises(RequestError):
+            await ft.explaincli(glide_client, str(uuid.uuid4()), "*")
+
+    async def _create_test_index_for_ft_explain_commands(
+        self, glide_client: GlideClusterClient, index_name: TEncodable
+    ):
+        # Helper function used for creating an index having hash data type, one text field and one numeric field.
+        fields: List[Field] = []
+        numeric_field: NumericField = NumericField("price")
+        text_field: TextField = TextField("title")
+        fields.append(text_field)
+        fields.append(numeric_field)
+
+        prefix = "{hash-search-" + str(uuid.uuid4()) + "}:"
+        prefixes: List[TEncodable] = []
+        prefixes.append(prefix)
+
+        assert (
+            await ft.create(
+                glide_client,
+                index_name,
```

Comment:
instead of index_name can you change it to like indexName1 for future indices naming

Reply:
This is a separate function that will always create 1 index. So, this is okay for now.